### DISCUSSION TITLE
Is there any way to get the line number of the code which caused the task panic

### DISCUSSION THREAD

**Discussion by tongvexil** [tokio-rs/tokio#6636] [Q&A]
Hi folks,
I am a beginner rust and tokio user so I hope this is the right place to ask my question.
I am trying to get more information from the task when it's panicking.
What I really want to know the line number which caused the panic but I don't know how to get it from the tokio::task::JoinError
With below code, my println! only prints out `Task panicked with info: Any { .. }`
Is there any way to get the exactly line number which caused panic and print it out using println!?
```
#[tokio::main]
async fn main() {
    let handle = tokio::spawn(async {
        panic!("Panicking in the task");
    });    

    let result = handle.await;
    match result {
        Ok(_) => {}
        Err(e) => {
            if e.is_panic() {
                println!("Task panicked with info: {:?}", e.into_panic());
            }
        }   
    }
}
```

**Comment by mox692** [comment]
> Is there any way to get the exactly line number

I think you could use custom panic handler by using [set_hook](https://doc.rust-lang.org/std/panic/fn.set_hook.html).

```rust
use std::panic;

#[tokio::main]
async fn main() {
    panic::set_hook(Box::new(|panic_info| {
        if let Some(location) = panic_info.location() {
            println!("panic occurred in file '{:?}'", location);
        } else {
            println!("panic occurred but can't get location information...");
        }
    }));
    
    let handle = tokio::spawn(async {
        panic!("Panicking in the task");
    });    

    let result = handle.await;
    
    match result {
        Ok(_) => {}
        Err(e) => {
            if e.is_panic() {
                println!("Task panicked with info: {:?}", e);
            }
        }   
    }
}
```


EDIT: It seems that you got [an answer in discord](https://discord.com/channels/500028886025895936/500336333500448798/1251165895204667412) as well.

**Reply by tongvexil** [reply]
Hi @mox692, 
Thanks for your response, the `panic::set_hook` works as expected but it does not contain information about the tokio task which panicked.
My goal is to get panic information from the tokio task to see which line of code caused the task panic. I need something like this: `println!("Task {:?} panicked at file: {:?} and line {:?}", task_info, panic.location.file(), panic.location.line())` with the task_info is the `JoinHandle` returned by `tokio::spawn`. Do you have any suggestion to archive this?

Thank you.
